File: mobile/lib/features/quiz/domain/entities/quiz.dart
import 'package:equatable/equatable.dart';

import '../../../features.dart';

class Quiz extends Equatable {
  final String id;
  final String courseId;
  final List<String> chapterIds;
  final String userId;
  final String name;
  final List<String>? questionIds;
  final List<Question>? questions;
  final int score;
  final bool isComplete;

  const Quiz({
    required this.id,
    required this.courseId,
    required this.chapterIds,
    required this.userId,
    required this.name,
    required this.questionIds,
    required this.questions,
    required this.score,
    required this.isComplete,
  });

  @override
  List<Object?> get props => [
        id,
        courseId,
        chapterIds,
        userId,
        name,
        questionIds,
        questions,
        score,
        isComplete,
      ];
}

File: new_backend/src/models/quiz.ts
import { required } from "joi";
import { Document, Schema, model } from "mongoose";

  export interface IQuiz extends Document {
    //   courseId: Schema.Types.ObjectId;
      name: String;
      questions: [Schema.Types.ObjectId];
      courseId: Schema.Types.ObjectId;
      userId: Schema.Types.ObjectId;
      chapters: [Schema.Types.ObjectId];
    }
    
    const QuizSchema: Schema<IQuiz> = new Schema({
      courseId:{
        type:Schema.Types.ObjectId,
        ref:'Course',
        required:[true,"chapter Id is required"]
      },
      chapters:{
        type: [{type:Schema.Types.ObjectId, ref: "Chapter"}],
        required: [true,"chapters are required"]
      },
      userId:{
        type:Schema.Types.ObjectId,
        ref:'User',
        required:[true,"user Id is required"]
      },
      name:{
        type: String,
        default:"Quiz"
      },
      questions:{
        type: [{type:Schema.Types.ObjectId, ref: "Question"}],
        required:[true,"questions is required to create quiz."],
        default: []
      }
    },
    {
        timestamps: {
          createdAt: 'createdAt',
          updatedAt: 'updatedAt'
        }
      }
    );
    
    const Quiz = model<IQuiz>("Quiz", QuizSchema);
    
    export default Quiz;